docs/release-process.md

```diff
@@ -1,9 +1,17 @@
-# Release Process for NGINX Kubernetes Gateway
+# Releases
 
-NGINX Kubernetes Gateway uses semantic versioning for its releases. For more information see https://semver.org.
+NGINX Kubernetes Gateway (NKG) uses semantic versioning for its releases. For more information see https://semver.org.
 
 Warning: Major version zero (0.y.z) is reserved for development, anything MAY change at any time. The public API is not stable.
 
+NKG, especially in its early stages, will release periodically on a feature set cadence. By feature set, we intend to mean 2 or 3 features as tracked by the repository's Github issues. We do this to remain agile and to expose features relatively quickly, but also to minimize the cost of many frequent releases and the associated process overhead.
+
+This means users can expect a release generally every 1 or 2 months, and can also track the progress of a release and its features by viewing our Github issues. Each issue scheduled for a release will be labelled with "proposal" and its scheduled release, the label will use the form vMAJOR.MINOR. Multiple issues may be written to describe one use-case and each will be labelled according to this process. Sub-issues should also link to their parent (the parent issue will list the links via mention notifications).
```

Comment:
label "enhancement" will be more appropriate for planned featured. "Proposal" is for something that was proposed but not committed to be implemented yet.

Reply:
Good point, we don't have any issue management process worked out. That needs writing too.